src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/CreateCollectionCommandResolver.java

```diff
@@ -51,44 +54,119 @@ public Class<CreateCollectionCommand> getCommandClass() {
+            && (command.options().indexing().deny() == null
+                || command.options().indexing().deny().isEmpty()))) {
+          try {
+            indexing = objectMapper.writeValueAsString(command.options().indexing());
+          } catch (JsonProcessingException e) {
+            // This should never happen because the object is extracted from json request
+            throw new RuntimeException(e);
+          }
+        }
       }
-      final int vectorSize = command.options().vector().dimension();
-      if (vectorSize > documentLimitsConfig.maxVectorEmbeddingLength()) {
-        throw new JsonApiException(
-            ErrorCode.VECTOR_SEARCH_FIELD_TOO_BIG,
-            String.format(
-                "%s: %d (max %d)",
-                ErrorCode.VECTOR_SEARCH_FIELD_TOO_BIG.getMessage(),
-                vectorSize,
-                documentLimitsConfig.maxVectorEmbeddingLength()));
+
+      // handling vector and vectorize options
+      if (command.options().vector() != null) {
+        if (!dataStoreConfig.vectorSearchEnabled()) {
+          throw new JsonApiException(
+              ErrorCode.VECTOR_SEARCH_NOT_AVAILABLE,
+              ErrorCode.VECTOR_SEARCH_NOT_AVAILABLE.getMessage());
+        }
+        function = command.options().vector().metric();
+        vectorSize = command.options().vector().dimension();
+        if (vectorSize > documentLimitsConfig.maxVectorEmbeddingLength()) {
+          throw new JsonApiException(
+              ErrorCode.VECTOR_SEARCH_FIELD_TOO_BIG,
+              String.format(
+                  "%s: %d (max %d)",
+                  ErrorCode.VECTOR_SEARCH_FIELD_TOO_BIG.getMessage(),
+                  vectorSize,
+                  documentLimitsConfig.maxVectorEmbeddingLength()));
+        }
+        if (command.options().vectorize() != null) {
+          try {
+            vectorize = objectMapper.writeValueAsString(command.options().vectorize());
+          } catch (JsonProcessingException e) {
+            // This should never happen because the object is extracted from json request
+            throw new RuntimeException(e);
+          }
+        }
       }
-      String vectorize = null;
-      if (command.options().vectorize() != null) {
+
+      String comment = null;
+      if (indexing != null || vectorize != null) {
+        final ObjectNode objectNode = objectMapper.createObjectNode();
         try {
-          vectorize = objectMapper.writeValueAsString(command.options().vectorize());
+          if (indexing != null) {
+            objectNode.put("indexing", objectMapper.readTree(indexing));
+          }
+          if (vectorize != null) {
+            objectNode.put("vectorize", objectMapper.readTree(vectorize));
+          }
         } catch (JsonProcessingException e) {
-          // This should never happen because the object is extracted from json request
           throw new RuntimeException(e);
         }
+
+        comment = objectNode.toString();
```

Comment:
Is there a reason we don't just use original JSON string?

Reply:
Original json is deserialized to object., here we needed the config as json to be stored in the comment

Reply:
Right but isn't it the same as original json String? So is there need to encode from Object again? It seemed like we were carrying json String along. If not, that's fine, not really problematic wrt performance (collection creation has many other costs).

Reply:
Ah ok. I finally got it: JSON is bound by Jackson into pieces, but then we need to store the whole JSON into Table comment. So we cannot avoid parts.

However: I think we can avoid going back and forth multiple times: but I understand why code right now does it -- because `command.options().indexing()` is not `JsonNode`.
But you can actually add all kinds of POJOs (anything Jackson can serialize) with:

```
objectNode.put("indexing", objectMapper.readTree(indexing));

// can be replaced with:

objectNode.putPOJO("indexing", command.options.indexing());
```

which should work if I understand code correctly. If I missed something else, feel free to ignore it.

Reply:
Made the change. Thanks for pointing out `objectNode.putPOJO`